### PR TITLE
Fix memory checking issue

### PR DESCRIPTION
**Pull request by qingwangrh**
Skip memory checking on vm_mem_limit setting
ID:1908606
Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
Passed test
python ConfigTest.py --testcase=physical_resources_check.i440fx --iothread_scheme=roundrobin --nr_iothreads=2 --platform=x86_64 --guestname=Win2019 --driveformat=virtio_scsi --nicmodel=virtio_net --imageformat=qcow2 --machines=i440fx --customsparams="vm_mem_limit = 12G\nimage_aio=native\ncd_format=ide" --clone=no

**Comment by qingwangrh** [comment]
@YongxueHong @zhenyzha Please help to review. Thanks.

File: qemu/tests/physical_resources_check.py [hunk 1]
```diff
@@ -291,13 +291,17 @@ def verify_machine_type():
...
-    if actual_mem != expected_mem:
-        fail_log = "Memory size mismatch:\n"
-        fail_log += "    Assigned to VM: %s\n" % expected_mem
-        fail_log += "    Reported by OS: %s\n" % actual_mem
-        n_fail.append(fail_log)
-        logging.error(fail_log)
+    if vm_mem_limit:
+        error_context.context("Skip memory checking %s" % vm_mem_limit, logging.info)
+    else:
+        if actual_mem != expected_mem:
```
[the remark this answers is not in the document]
Comment (qingwangrh) [reply qemu/tests/physical_resources_check.py:299]: Thanks,updated

**Review by YongxueHong — APPROVED** [review]
LGTM.

**Comment by zhenyzha** [comment]
this case cfg set `only i386, x86_64`
So please ignore me

**Comment by qingwangrh** [comment]
@menli820 please help to review, thanks.

**Comment by menli820** [comment]
LGTM

PASS 2-Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.physical_resources_check

**Pull request merged**